src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java

```diff
@@ -104,5 +129,54 @@ void shouldThrowAnExceptionWhenNameContainsIllegalLetters() throws Exception {
         result.andExpect(status().isBadRequest());
     }
 
+    @Test
+    @DisplayName("Should return empty list of wallets and http ok status")
+    void shouldReturnEmptyList() throws Exception {
+        // when
+        MockHttpServletResponse result = mockMvc.perform(get("/api/wallet")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn()
+                .getResponse();
+
+        // then
+        assertThat(result.getStatus()).isEqualTo(200);
+        assertThat(result.getContentAsString()).isEqualTo("[]");
+        assertThat(result.getHeader("message")).isEqualTo("There are no available wallets to view.");
+    }
+
+    @Test
+    @DisplayName("Should return information about all wallets and http ok status")
+    void shouldReturnListOfAllWallets() throws Exception {
+        // given
+        List<WalletDTO> listOfWalletsDTO = createListOfWalletsDTO();
+        given(walletService.getAllWallets()).willReturn(listOfWalletsDTO);
+
+        // then
+        mockMvc.perform(get("/api/wallet")
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(header().string("message", "The list of wallets has been successfully retrieved."))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(3))
```

Comment:
ups... value '3' is not a constant - and rest of values are.
In other words maybe it's not good to create constants from variables. Specially when they are used only in one method of the test. And you created constants only from part of them. 

**Second thing is:**
Compilers optimize constants by replacing them inline, 
which will make changing the constant useless. For example,
assume changeField is our method which changes the static final field TEST of Class clazz:

```java
public static final String TEST = "Hello";

public static void main( String[] args )
{
    System.out.println( TEST ); // prints "Hello"
    changeField( clazz, "TEST", "hi!" );
    System.out.println( TEST ); // prints "Hello"
}

This is because during compilation, your compiler decided to replace
    System.out.println( TEST ); 
with 
    System.out.println( "Hello" ); 
```

It will be good to read about field constant convention used in tests. Trust me, the test development tutorial is also under construction. But not everything is included in it and properly formatted to make it readable.
**My suggestion for now is:** don't use it here at all. One test class, one test method and 12 (or more) constants - and all used in exactly one place one-by-one.

Reply:
- [ ] change method of delivery expected size result

Reply:
Used `walletList.size()` insted of `3` int value